src/server/game/Entities/Player/Player.cpp

```diff
@@ -686,6 +686,10 @@ bool Player::Create(ObjectGuid::LowType guidlow, CharacterCreateInfo* createInfo
     }
     // all item positions resolved
 
+    // ensure player starts with full health
+    UpdateAllStats();
+    SetFullHealth();
```

Comment:
https://github.com/azerothcore/azerothcore-wotlk/blob/508b1658fcbc4e625f939da75963480be2354430/src/server/game/Entities/Player/Player.cpp#L589-L591

We already do this in Player::Create

Reply:
Yes but it has do be re-done after equipping, as the equipped items at least for DK change the stats.
I thought about removing the first call that you linked but was unsure about side effects, so I just added a second call at the end to make sure.

Reply:
Remove the first in that case

Reply:
Doubt it can have any side effects